konnectivity-client/pkg/client/client.go

```diff
@@ -76,6 +76,7 @@ func CreateSingleUseGrpcTunnel(ctx context.Context, address string, opts ...grpc
 
 	stream, err := grpcClient.Proxy(ctx)
 	if err != nil {
+		c.Close()
```

Comment:
`c.Close()` was only called from tunnel.serve() below. If grpc.Client.Proxy() errors, which is possible when there are no client connections available, we would never call c.Close()

Reply:
good catch